src/ja/server/scheduler/algorithm.py

```diff
@@ -0,0 +1,115 @@
+from abc import ABC, abstractmethod
+from typing import List, Optional, Tuple
+from ja.common.job import Job
+from ja.server.database.database import ServerDatabase
+from ja.server.database.types.work_machine import WorkMachine
+
+
+class CostFunction(ABC):
+    """
+    A CostFunction is a collection of functions which determine the order in
+    which jobs are scheduled.
+    """
+
+    @abstractmethod
+    def calculate_cost(self, job: Job) -> int:
```

Comment:
What about the parameters for the cost function?
Are you going to pass them to the CostFunction constructor?

Reply:
CostFunction is an abstract class. Any specific constants belong to the concrete implementation of that cost function.